alpha/apps/kaltura/modules/kmcng/actions/kmcngAction.class.php

```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * @package    Core
+ * @subpackage KMCNG
+ */
+class kmcngAction extends kalturaAction
+{
+	const SYSTEM_DEFAULT_PARTNER = 0;
+	
+	public function execute ( ) 
+	{
+		sfView::SUCCESS;
+
+		$kmcngParams = kConf::get('kmcng');
+
+		if (!$kmcngParams)
+		{
+			KalturaLog::warning("kmcng config doesn't exist in configuration.");
+			return sfView::ERROR;
+		}
+
+		// Check for forced HTTPS
+		if( (!$kmcngParams["kmcng_debug_mode"]) )
+		{
+			if ((!isset($_SERVER['HTTPS']) || $_SERVER['HTTPS'] != 'on'))
+			{
+				header("Location: " . infraRequestUtils::PROTOCOL_HTTPS . "://" . $_SERVER['SERVER_NAME'] . $_SERVER["REQUEST_URI"]);
+				die();
+			}
+			header("Strict-Transport-Security: max-age=63072000; includeSubdomains; preload");
+		}
+
+		//disable cache
+		header("Cache-Control: no-store, no-cache, must-revalidate, max-age=0");
+		header("Cache-Control: post-check=0, pre-check=0", false);
+		header("Pragma: no-cache");
+
+		if ( !$kmcngParams["kmcng_version"])
+		{
+			KalturaLog::warning("kmcng_version doesn't exist in configuration.");
+			return sfView::ERROR;
+		}
+
+		try
```

Comment:
None of the functions within this block throws exception on failure , try catch is useless , please remove